### PR TITLE
chore: add DENO_FUTURE env var

### PR DESCRIPTION
**Pull request by littledivy**
Closes https://github.com/denoland/deno/issues/22315

```
~> DENO_FUTURE=1 target/debug/deno

> globalThis.window
undefined
```

### CODE REVIEW

File: cli/worker.rs [hunk 1]
```diff
@@ -596,6 +596,7 @@ impl CliMainWorkerFactory {
         node_ipc_fd: shared.node_ipc,
         disable_deprecated_api_warning: shared.disable_deprecated_api_warning,
         verbose_deprecated_api_warning: shared.verbose_deprecated_api_warning,
+        future: std::env::var("DENO_FUTURE").is_ok(),
```
Comment (bartlomieju) [inline cli/worker.rs:599]: This env var should be captured in `cli/args/mod.rs` and exposed as a method on `CliOptions` struct - maybe `CliOptions::enable_future_features()`?

**Review by bartlomieju — COMMENTED** [review]
Please add an integration test

File: cli/args/mod.rs [hunk 2]
```diff
@@ -987,6 +987,10 @@ impl CliOptions {
     }
   }
 
+  pub fn enable_future_features(&self) -> bool {
+    std::env::var("DENO_FUTURE").is_ok()
```
Comment (satyarohith) [inline cli/args/mod.rs:991]: maybe we should ensure that the parsed value results in boolean true?

Comment (littledivy) [reply cli/args/mod.rs:991]: we only want to check for presence of the env. eg: this should work `DENO_FUTURE=1`

Comment (satyarohith) [reply cli/args/mod.rs:991]: `DENO_FUTURE=false` will also enable future features even though user explicitly disabled it.

Looks like there is inconsistency in the codebase right now:
1. https://github.com/denoland/deno/blob/83d72e5c1c0b983f4b63f3f64f9a1de7600985f4/cli/tools/jupyter/mod.rs#L54-L57
2. https://github.com/denoland/deno/blob/1ad754b4123009e01dbecb3b880e7f0545e46c2f/cli/lsp/language_server.rs#L1148-L1152

**Review by satyarohith — APPROVED** [review]
LGTM apart from the nit

File: cli/args/mod.rs (hunk 2 above)
Comment (bartlomieju) [reply cli/args/mod.rs:991]: Additionally reading of env vars should be done only once in `Lazy` at the top level. With this setup we will read env var on each invocation of this function.

**Pull request merged**